### PR TITLE
Exercise 2

### PR DESCRIPTION
**Pull request by Wiesje**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -114,3 +114,6 @@ env/
 
 ### Configuration files ###
 .env
+
+### Secret files ###
+secrets.py
```
Comment (Jonesey13) [inline .gitignore:119]: Probably worth explaining how to set this file up in the README

File: LinkBuilder.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import secrets
+
+class LinkBuilder:
+    baseUrl = "https://api.trello.com"
```
Comment (Jonesey13) [inline LinkBuilder.py:4]: Might want to add `/1/` to the base url here

File: LinkBuilder.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    yourKey = secrets.TRELLO_API_KEY
+    yourToken = secrets.TRELLO_API_TOKEN
+
+    def __init__(self):
+        self.to_do_list = "5eeb6d36261b73280b211344"
```
Comment (Jonesey13) [inline LinkBuilder.py:9]: These 2 ids can be inferred from the list names.

File: LinkBuilder.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        self.done_list = "5eec7a03ebdf3b6d3eb2158e"
+        self.index_link = f"{self.baseUrl}/1/lists/{self.to_do_list}/cards?key={self.yourKey}&token={self.yourToken}"
```
Comment (Jonesey13) [inline LinkBuilder.py:11]: Is there a way to avoid repeating the key and token every time?

File: LinkBuilder.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import secrets
+
+class LinkBuilder:
```
Comment (Jonesey13) [inline LinkBuilder.py:3]: The requests library supports providing query params so putting everything in a link builder might not be the clearest 
abstraction here (I'd probably create something more specific like a `TrelloApi` class with `@staticmethod`s).

File: app.py [hunk 6]
```diff
@@ -1,18 +1,35 @@
 from flask import Flask, render_template, request, redirect, url_for
 import session_items as session
```
Comment (Jonesey13) [inline app.py:2]: We're no longer using this file (also the `SECRET_KEY` config value is now obsolete as it manages the session cookie)

File: Item.py [hunk 7]
```diff
@@ -0,0 +1,4 @@
+class Item:
```
Comment (Jonesey13) [inline Item.py:1]: This class is never used

File: app.py [hunk 8]
```diff
@@ -1,18 +1,35 @@
...
+import requests
+import LinkBuilder
 
 app = Flask(__name__)
 app.config.from_object('flask_config.Config')
```
Comment (Jonesey13) [inline app.py:7]: This line can now be removed

File: app.py [hunk 9]
```diff
@@ -1,18 +1,35 @@
...
 
+link_builder = LinkBuilder.LinkBuilder()
+
 @app.route('/')
 def index():
-    item_list = session.get_items()
+    r = requests.get(link_builder.get_index_link())
+    item_list = r.json()
```
Comment (Jonesey13) [inline app.py:14]: We should be using a todo item class somewhere here.

File: requirements.txt [hunk 10]
```diff
@@ -1,2 +1,12 @@
+certifi==2020.4.5.2
```
Comment (Jonesey13) [inline requirements.txt:1]: Do we use all these dependencies?

Comment (Wiesje) [reply requirements.txt:1]: I am not sure, how can I check this?

Comment (Jonesey13) [reply requirements.txt:1]: I'm guessing they'd be referred to by an `import` statement somewhere? (with the exception of maybe python-dotenv)

File: README.md [hunk 11]
```diff
@@ -29,3 +29,10 @@ You should see output similar to the following:
  * Debugger PIN: 226-556-590
 ```
 Now visit [`http://localhost:5000/`](http://localhost:5000/) in your web browser to view the app.
+
+## Setting up secrets.py
+To link the app to your own Trello board, create an API key and token [here](https://trello.com/app-key) and paste them in a file called secrets.py which should look like this:
+```python
+TRELLO_API_KEY = "XXXXXXXXXXXXX"
```
Comment (Jonesey13) [inline README.md:36]: This wouldn't be enough to set this up from scratch as they'd need to provide list ids

File: LinkBuilder.py [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+    baseUrl = "https://api.trello.com/1"
+    yourKey = secrets.TRELLO_API_KEY
+    yourToken = secrets.TRELLO_API_TOKEN
+
+    def __init__(self):
+        self.to_do_list = "5eeb6d36261b73280b211344"
+        self.done_list = "5eec7a03ebdf3b6d3eb2158e"
+        self.index_link = f"{self.baseUrl}/lists/{self.to_do_list}/cards?key={self.yourKey}&token={self.yourToken}"
```
Comment (Jonesey13) [inline LinkBuilder.py:11]: I'd argue that you'd probably want a mechanism to show completed todos as well

**Comment by Jonesey13** [comment]
@Wiesje Could you please re-submit this to Aptem so I can approve this project exercise?

**Pull request merged**